### PR TITLE
Update gcn2conv.py

### LINKED ISSUE #5975: [Bug] An useless assignment in `GCN2Conv`
## 🐛 Bug

https://github.com/dmlc/dgl/blob/e178158616012f2e3bc83e28457127ce6dc0cf03/python/dgl/nn/pytorch/conv/gcn2conv.py#L258-L267

The assignment of Line 259 seems to be useless and thus can be replaced by

```python3
feat.add_(feat_0)
```

Or Line 260 mistyped `=` instead of `+=`?

## To Reproduce

1.
1.
1.

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
Simplifies the code in GCN2Conv.  Closes #5975.

### CODE REVIEW

File: python/dgl/nn/pytorch/conv/gcn2conv.py [hunk 1]
```diff
@@ -256,29 +256,12 @@ def forward(self, graph, feat, feat_0, edge_weight=None):
...
             if self._project_initial_features:
-                rst = feat.add_(feat_0)
-                rst = th.addmm(
-                    feat,
-                    feat,
-                    self.weight1,
-                    beta=(1 - self.beta),
-                    alpha=self.beta,
-                )
+                rst = feat + feat_0
```
Comment (frozenbugs) [inline python/dgl/nn/pytorch/conv/gcn2conv.py:259]: To keep consistency?
```suggestion
                feat_sum = feat + feat_0
```

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/nn/pytorch/conv/gcn2conv.py [hunk 2]
```diff
@@ -256,29 +256,12 @@ def forward(self, graph, feat, feat_0, edge_weight=None):
...
-                rst = feat.add_(feat_0)
-                rst = th.addmm(
-                    feat,
-                    feat,
-                    self.weight1,
-                    beta=(1 - self.beta),
-                    alpha=self.beta,
-                )
+                feat_sum = feat + feat_0
+                rst = (1 - self.beta) * rst + self.beta * rst @ self.weight1
```
Comment (jermainewang) [inline python/dgl/nn/pytorch/conv/gcn2conv.py:260]:
```suggestion
                rst = (1 - self.beta) * feat_sum + self.beta * feat_sum @ self.weight1
```

**Comment by jermainewang** [comment]
@frozenbugs updated according to your suggestion. Please check again.

File: python/dgl/nn/pytorch/conv/gcn2conv.py [hunk 3]
```diff
@@ -256,29 +256,12 @@ def forward(self, graph, feat, feat_0, edge_weight=None):
...
-                rst = feat.add_(feat_0)
-                rst = th.addmm(
-                    feat,
-                    feat,
-                    self.weight1,
-                    beta=(1 - self.beta),
-                    alpha=self.beta,
-                )
+                feat_sum = feat + feat_0
+                rst = (1 - self.beta) * feat_sum + self.beta * feat_sum @ self.weight1
```
Comment (frozenbugs) [inline python/dgl/nn/pytorch/conv/gcn2conv.py:260]:
```suggestion
                rst = ((1 - self.beta) * feat_sum
                       + self.beta * feat_sum @ self.weight1)
```

**Pull request merged**